AOT build throwing the error: Function calls are not supported in decorators but ‘FlexLayoutModule’ was called, when using withConfig options

# Bug Report

#### What is the expected behavior?
App should build in AOT when FlexLayoutModule.withConfig({addFlexToParent: false}) is added to the module.

#### What is the current behavior?
App fails to build in AOT with error:  Function calls are not supported in decorators but ‘FlexLayoutModule’ was called, when using withConfig options

#### What are the steps to reproduce?
include in Module: FlexLayoutModule.withConfig({addFlexToParent: false})


#### What is the use-case or motivation for changing an existing behavior?


#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 6.0.4, Material 6.2.1

#### Is there anything else we should know?
We have another module:     LocalStorageModule.withConfig({ prefix: '', storageType: 'localStorage' }) that builds fine.

I've also included it in your demo app, but unsure if it's building in AOT, we have another separate app, I added the same line to the app module, and still get this error: ERROR in Error during template compile of 'AppModule'
  Function calls are not supported in decorators but 'FlexLayoutModule' was called.
node_modules/@angular/flex-layout/typings/module.d.ts(12,37): error TS1183: An implementation cannot be declared in ambient contexts

--- a/src/apps/demo-app/src/app/app.module.ts
+++ b/src/apps/demo-app/src/app/app.module.ts
@@ -23,7 +23,7 @@ const PRINT_BREAKPOINTS = [{
     BrowserAnimationsModule,
     RoutingModule,
     DemoMaterialModule,
-    FlexLayoutModule,
+    FlexLayoutModule.withConfig({useColumnBasisZero: false}),
   ],
   providers: [{provide: BREAKPOINT, useValue: PRINT_BREAKPOINTS, multi: true}],
   bootstrap: [AppComponent]

--- a/src/lib/flex/flex/flex.ts
+++ b/src/lib/flex/flex/flex.ts
@@ -163,8 +163,9 @@ export class FlexDirective extends BaseDirective implements OnInit, OnChanges, O
   protected _validateValue(grow: number|string,
                            shrink: number|string,
                            basis: string|number|FlexBasisAlias) {
+    let addFlexToParent = this.layoutConfig.addFlexToParent !== false;
     // The flex-direction of this element's flex container. Defaults to 'row'.
-    let layout = this._getFlexFlowDirection(this.parentElement, this.layoutConfig.addFlexToParent);
+    let layout = this._getFlexFlowDirection(this.parentElement, addFlexToParent);
     let direction = (layout.indexOf('column') > -1) ? 'column' : 'row';
 
     let max = isFlowHorizontal(direction) ? 'max-width' : 'max-height';

--- a/src/lib/flex/flex/flex.ts
+++ b/src/lib/flex/flex/flex.ts
@@ -210,8 +210,8 @@ export class FlexDirective extends BaseDirective implements OnInit, OnChanges, O
     };
     switch (basis || '') {
       case '':
-        basis = direction === 'row' ? '0%' :
-          (this.layoutConfig.useColumnBasisZero ? '0.000000001px' : 'auto');
+        const useColumnBasisZero = this.layoutConfig.useColumnBasisZero !== false;
+        basis = direction === 'row' ? '0%' : (useColumnBasisZero ? '0.000000001px' : 'auto');
         break;
       case 'initial':   // default
       case 'nogrow':

--- a/src/lib/module.ts
+++ b/src/lib/module.ts
@@ -11,12 +11,10 @@ import {
   NgModule,
   Optional,
   PLATFORM_ID,
-  Provider,
 } from '@angular/core';
 import {isPlatformServer} from '@angular/common';
 import {
   SERVER_TOKEN,
-  DEFAULT_CONFIG,
   LayoutConfigOptions,
   LAYOUT_CONFIG,
   BreakPoint,
@@ -50,30 +48,26 @@ export class FlexLayoutModule {
    */
   static withConfig(configOptions: LayoutConfigOptions,
                     breakpoints?: BreakPoint|BreakPoint[]): ModuleWithProviders {
-    const config = Object.assign({}, DEFAULT_CONFIG);
-    const moduleProviders: Provider[] = [];
-
-    for (const key in configOptions) {
-      // If the setting is different and not undefined or null, change it
-      if (configOptions[key] !== config[key] &&
-        (configOptions[key] === false || configOptions[key] === true)) {
-        config[key] = configOptions[key];
-      }
-    }
-
-    if (configOptions.serverLoaded) {
-      moduleProviders.push({provide: SERVER_TOKEN, useValue: true});
-    }
-
-    if (Array.isArray(breakpoints)) {
-      moduleProviders.push({provide: BREAKPOINT, useValue: breakpoints, multi: true});
-    }
-
-    moduleProviders.push({provide: LAYOUT_CONFIG, useValue: config});
-
     return {
       ngModule: FlexLayoutModule,
-      providers: moduleProviders
+      providers: Array.isArray(breakpoints) ?
+        configOptions.serverLoaded ?
+          [
+            {provide: LAYOUT_CONFIG, useValue: configOptions},
+            {provide: BREAKPOINT, useValue: breakpoints, multi: true},
+            {provide: SERVER_TOKEN, useValue: true},
+          ] : [
+            {provide: LAYOUT_CONFIG, useValue: configOptions},
+            {provide: BREAKPOINT, useValue: breakpoints, multi: true},
+          ]
+        :
+        configOptions.serverLoaded ?
+          [
+            {provide: LAYOUT_CONFIG, useValue: configOptions},
+          ] :
+          [
+            {provide: LAYOUT_CONFIG, useValue: configOptions},
+          ]
     };
   }
 

--- a/src/lib/module.ts
+++ b/src/lib/module.ts
@@ -64,6 +64,7 @@ export class FlexLayoutModule {
         configOptions.serverLoaded ?
           [
             {provide: LAYOUT_CONFIG, useValue: configOptions},
+            {provide: SERVER_TOKEN, useValue: true},
           ] :
           [
             {provide: LAYOUT_CONFIG, useValue: configOptions},

